hashfs/hashfs.py

```diff
@@ -424,3 +454,62 @@ def close(self):
             self._obj.close()
         else:
             self._obj.seek(self._pos)
+
+
+class PutStrategies:
+    """Namespace for built-in put strategies.
+
+    Should not be instantiated. Use the :meth:`get` static method to look up a
+    strategy by name, or directly reference one of the included class methods.
+    """
+
+    @classmethod
+    def get(cls, method):
+        """Look up a stragegy by name string. You can also pass a function
+        which will be returned as is."""
+        if method:
+            if method == 'get':
+                raise ValueError("invalid put strategy name, 'get'")
+            return method if is_callable(method) else getattr(cls, method)
+
+    @staticmethod
+    def copy(hashfs, src_stream, dst_path):
+        """The default copy put strategy, writes the file object to a
+        temporary file on disk and then moves it into place."""
+        shutil.move(hashfs._mktempfile(src_stream), dst_path)
```

Comment:
The `_mktempfile()` method will likely need to be moved into this function or made standalone since the original `HashFS._copy` and `HashFS._mktempfile()` were only ever used together. So if `_copy()` is moving, makes sense to move `_mktempfile()`.

However, the `fmode` setting (which is currently done in `HashFS._mktempfile()`) should remain a part of `HashFS`. Basically, it should be the responsibility of `HashFS.put` to set `fmode` on the newly added files. That way user-land put strats don't need to worry about setting `fmode` themselves.